// @dexhunterio/swaps/lib/swap/components/SwapWarning.js
import { jsx as r, Fragment as e, jsxs as s } from "react/jsx-runtime";
import d from "../../components/ui/tooltipDialog.js";
import { a as m } from "../../axios-ddd885c5.js";
import { useMemo as p } from "react";
import { cn as n } from "../../lib/utils.js";
import { u as h } from "../../useQuery-714c7c61.js";
import "../../hooks/useScreen.js";
import "../../components/ui/dialog.js";
import "../../index-840f2930.js";
import "../../index-1c873780.js";
import "../../index-c7156e07.js";
import "../../index-563d1ed8.js";
import "../../index-4914f99c.js";
import "../../index-67500cd3.js";
import "../../index-c8f2666b.js";
import "../../_commonjsHelpers-10dfc225.js";
import "../../index-27cadef5.js";
import "../../index-5116e957.js";
import "../../lib.js";
import "../../extend-tailwind-merge-e63b2b56.js";
import "../../components/ui/tooltip.js";
import "../../index-0ce202b9.js";
import "../../index-bcfeaad9.js";
import "../../index-f7426637.js";
import "../../index-ca8eb9e1.js";
import "../../index-9d43cc8b.js";
import "../../QueryClientProvider-103c6f38.js";
import "../../notifyManager-b5b16d35.js";
const A = () => {
  const { data: i, isLoading: a } = h(
    ["dh-chain-load"],
    async () => {
      try {
        const { data: t } = await m.get("https://pool.pm/total.json");
        return Math.round(
          Math.max(t == null ? void 0 : t.load_1h, t == null ? void 0 : t.load_5m) * 100
        );
      } catch {
        return 0;
      }
    },
    {
      refetchInterval: 1e4
    }
  ), o = p(() => i || 0, [i]);
  return /* @__PURE__ */ r(e, { children: !a && o >= 90 && /* @__PURE__ */ r("div", { className: "dhs-flex dhs-w-full dhs-py-[2px] dhs-px-5 dhs-justify-center dhs-items-center dhs-gap-2 dhs-h-[30px] dhs-font-proximaBold dhs-text-warning", children: /* @__PURE__ */ r(
    d,
    {
      trigger: /* @__PURE__ */ r(
        "div",
        {
          className: n(
            "dhs-flex dhs-rounded-[15px] dhs-hover:bg-gray-109 dhs-hover:bg-opacity-30 dhs-text-accent dhs-font-proximaBold dhs-gap-1 dhs-items-center dhs-text-md",
            o >= 90 && "dhs-text-red-101"
          ),
          children: /* @__PURE__ */ s("span", { children: [
            "High chain load at ",
            o.toFixed(0),
            "%. Swaps may take longer."
          ] })
        }
      ),
      content: `Cardano chain load is ${o.toFixed(0)}%`,
      contentClass: "dhs-text-white"
    }
  ) }) });
};
export {
  A as default
};
