// @dexhunterio/swaps/lib/constants/dexes.js
const e = {
  SUNDAESWAPV3: {
    name: "SundaeSwap V3",
    isEnabled: !0,
    url: "https://sundaeswap.finance/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/sundaev3.webp",
    code: "SUNDAESWAPV3",
    image: !0,
    isStopLossEnabled: !1,
    isDCAEnabled: !1,
    isV2: !0
  },
  SPLASH: {
    isEnabled: !0,
    name: "Splash",
    url: "https://splash.trade/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/splashlogo.jpeg",
    code: "SPLASH",
    image: !0,
    isStopLossEnabled: !1,
    isDCAEnabled: !1,
    isV2: !0
  },
  MINSWAPV2: {
    name: "Minswap V2",
    isEnabled: !0,
    url: "https://app.geniusyield.co/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/minswapv2.webp",
    code: "MINSWAPV2",
    image: !0,
    isStopLossEnabled: !1,
    isDCAEnabled: !1,
    isV2: !0
  },
  MINSWAP: {
    name: "MinSwap V1",
    isEnabled: !0,
    url: "https://minswap.org/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/minswap.png",
    code: "MINSWAP",
    image: !0,
    isStopLossEnabled: !0,
    isDCAEnabled: !0,
    isV2: !1
  },
  AXO: {
    name: "Axo Trade",
    isEnabled: !0,
    url: "https://axo.trade/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/axo.jpeg",
    code: "AXO",
    image: !0,
    isStopLossEnabled: !1,
    isDCAEnabled: !1,
    isV2: !0
  },
  SPECTRUM: {
    name: "Spectrum",
    isEnabled: !0,
    url: "https://spectrum.cards/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/spectrum.png",
    code: "SPECTRUM",
    isStopLossEnabled: !0,
    isDCAEnabled: !0,
    isV2: !0
  },
  SUNDAESWAP: {
    name: "SundaeSwap V1",
    isEnabled: !0,
    url: "https://sundaeswap.finance/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/sundae.png",
    code: "SUNDAESWAP",
    isStopLossEnabled: !0,
    isDCAEnabled: !0,
    isV2: !1
  },
  WINGRIDER: {
    name: "Wingriders",
    isEnabled: !0,
    url: "https://wingriders.io/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/wingriders.png",
    code: "WINGRIDER",
    image: !0,
    isStopLossEnabled: !0,
    isDCAEnabled: !0,
    isV2: !1
  },
  VYFI: {
    name: "VyFinance",
    isEnabled: !0,
    url: "https://vy.finance/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/vyfi.png",
    code: "VYFI",
    image: !0,
    isStopLossEnabled: !0,
    isDCAEnabled: !0,
    isV2: !1
  },
  CERRA: {
    name: "Cerra",
    isEnabled: !0,
    url: "https://app.cerra.io",
    logo: "https://storage.googleapis.com/dexhunter-images/public/cerra3.svg",
    code: "CERRA",
    isStopLossEnabled: !1,
    isDCAEnabled: !1,
    isV2: !1
  },
  MUESLISWAP: {
    name: "MuesliSwap",
    isEnabled: !0,
    url: "https://muesliswap.com/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/mueslidex.svg",
    code: "MUESLISWAP",
    image: !1,
    isStopLossEnabled: !0,
    isDCAEnabled: !0,
    isV2: !1
  },
  TEDDY: {
    name: "TeddySwap",
    isEnabled: !0,
    url: "https://teddyswap.org/",
    logo: "https://storage.googleapis.com/dexhunter-images/public/teddyswap.jpeg",
    code: "TEDDY",
    image: !1,
    isStopLossEnabled: !1,
    isDCAEnabled: !1,
    isV2: !1
  }
  // GENIUS: {
  //   name: 'Genius Yield',
  //   url: 'https://app.geniusyield.co/',
  //   logo: "https://storage.googleapis.com/dexhunter-images/public/geniusyield.jpeg",
  //   code: 'GENIUS'
  // },
}, s = {
  name: "DexHunter",
  url: "https://dexhunter.io/",
  logo: "https://storage.googleapis.com/dexhunter-images/public/coin.svg",
  code: "DEXHUNTER"
};
export {
  s as DEXHUNTER_DEX_INFO,
  e as dexes
};
